src/Raven.Server/Documents/Indexes/Auto/AutoMapIndex.cs

```diff
@@ -70,12 +71,16 @@ private void ExecuteCleanup(CancellationToken token, DocumentsOperationContext d
                 var count = 0;
 
                 var sw = Stopwatch.StartNew();
-                using (var indexActions = IndexPersistence.OpenIndexWriter(indexContext.Transaction.InnerTransaction))
+                IndexWriteOperation indexWriter = null;
+                try
                 {
                     using (databaseContext.OpenReadTransaction())
                     {
                         foreach (var tombstone in DocumentDatabase.DocumentsStorage.GetTombstonesAfter(databaseContext, collection, lastEtag + 1, 0, pageSize))
                         {
+                            if (indexWriter == null)
```

Comment:
Doesn't this mean that we are creating / disposing the index writer all the time?
IIRC, we got better perf by having a cached instance of the index writer and recreate it every now and then